src/main/java/org/dataloader/DataLoaderHelper.java

```diff
@@ -98,34 +99,74 @@ Object getCallContext() {
         return Optional.empty();
     }
 
+    /**
+     * Try to get a value from the cache.
+     * Returns empty if cache doesn't contain key, or get completes with null or exception.
+     *
+     * @param key - key to fetch
+     * @return optional value
+     */
+    CompletableFuture<Optional<V>> loadFromCache(K key) {
+        boolean cachingEnabled = loaderOptions.cachingEnabled();
+
+        if (!cachingEnabled) {
+            return CompletableFuture.completedFuture(Optional.empty());
+        }
+
+        if (!futureCache.containsKey(key)) {
+            return CompletableFuture.completedFuture(Optional.empty());
+        }
+
+        return futureCache.get(key).handleAsync((v, exception) -> {
```

Comment:
`handleAsync` use the standard ForkJoin pool - we make it a policy to never introduce threading into the library.  Only the consuming code of this lib can really know what threading strategy to use.

If we accepted this change, this would have to be the version that takes and `executor` and it would have to come from DataLoaderOptions say